package.json

```diff
@@ -30,8 +30,8 @@
     "url": "https://github.com/andig/pimatic-fritz"
   },
   "dependencies": {
-    "smartfritz-promise": "^0.4.1",
```

Comment:
That's not correct. Changing it to ^0.4.2 won't install 0.5.0! According to semver, the caret  "Allows changes that do not modify the left-most non-zero digit in the [major, minor, patch] tuple. ". So, in the given case it will only install newer maint. release, e.g. 0.4.3.

See also https://docs.npmjs.com/misc/semver#caret-ranges-123-025-004

Regarding the issues I had with 0.5 I'll get back to you with more information.